pkg/systemd/quadlet/quadlet.go

```diff
@@ -1650,6 +1651,11 @@ func ConvertPod(podUnit *parser.UnitFile, name string, unitsInfoMap map[string]*
 	execStartPre.add("--infra-name", fmt.Sprintf("%s-infra", podName))
 	execStartPre.add("--name", podName)
 
+	shmSize := podUnit.LookupAll(PodGroup, KeyShmSize)
```

Comment:
IIUC you are looking for all the occurrences of `KeyShmSize` and then use the first one. This does not align with the expected behaviour in which the last value takes precedence.
Instead of calling `LookupAll` and using the first element of the array, you should use `Lookup` (which returns the last occurrence) and use the returned value if exists.
Alternatively, you can create a map and use the `lookupAndAddString` function. Yes, it is an overkill for a single key, but it's the start for supporting more keys

Reply:
I changed from `LookupAll` to  `Lookup`, I tried a to implement the map using the `stringsKeys` that is already created and it failed on my e2e assertions tests, tried also to create a new map just for sanity and it failed as well, if we really need to use map I can give a try later this week.

Reply:
I added a comment on the new code. We'll change the code to use a map later

Reply:
The map problem was in my e2e test, instead of using `assert-podman-pre-args "--shm-size=5g"` I'm now using `assert-podman-pre-args "--shm-size" "5g"` this made the test pass.

I also changed the `stringKeys` function from `lookupAndAddAllStrings` to `lookupAndAddString`.